### PR TITLE
SUBMARINE-824. Execute port-forwarding automatically with Golang out-of-cluster

### PR DESCRIPTION
**Pull request by kevin85421**
### What is this PR for?
Execute the following command via Golang out-of-cluster
```
kubectl port-forward --address 0.0.0.0 -n ${Your_Namespace} service/traefik 32080:80
```

Reference: 
(1) https://gianarb.it/blog/programmatically-kube-port-forward-in-go
(2) https://github.com/minio/operator/blob/master/kubectl-minio/cmd/proxy.go

### What type of PR is it?
[Feature]

### Todos
* Execute port-forwarding automatically with Golang in-cluster
     *  The function `k8sutil.ServicePortForwardPort(context.TODO(), namespace, "traefik", 32080, 80, color.FgGreen)` does not work in-cluster.
     * (In submarine-operator pod) `curl 127.0.0.1:32080` --> Yes
     * (Local Chrome browser) 127.0.0.1:32080 --> 404
     * We need to forward the request from 127.0.0.1:32080 to the submarine-operator pod. (Use k8s NodePort to implement)

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-824

### How should this be tested?
https://user-images.githubusercontent.com/20109646/118227081-86001080-b4ba-11eb-9274-6a053954f5f7.mov

### Questions:
* Do the license files need updating? No
* Are there breaking changes for older versions? No
* Does this need new documentation? No

### CODE REVIEW

**Comment by kevin85421** [comment]
@Kenchu123 @MortalHappiness can you help me review this PR? Thanks!

File: submarine-cloud-v2/controller.go [hunk 1]
```diff
@@ -112,6 +115,7 @@ type WorkQueueItem struct {
 
 // NewController returns a new sample controller
 func NewController(
+	incluster bool,
```
Comment (MortalHappiness) [inline submarine-cloud-v2/controller.go:118]: Here you pass `incluster` as parameter but not used.

File: submarine-cloud-v2/controller.go [hunk 2]
```diff
@@ -1134,6 +1136,17 @@ func (c *Controller) syncHandler(workqueueItem WorkQueueItem) error {
...
 			return err
 		}
+
+		// Port-forwarding
+		// TODO:
+		//   (1) multi-tenant port-forwarding
+		//   (2) Basic operations: on/off/modify (change port)
+		//   (3) in-cluster
+		if action == ADD {
+			if !incluster {
```
Comment (MortalHappiness) [inline submarine-cloud-v2/controller.go:1146]: Here the `incluster` still works. But the `incluster` here is actually the global variable in `main.go`. I think what you want to do is to add this `incluster` as one attribute of `struct Controller` and use it.

File: submarine-cloud-v2/controller.go [hunk 3]
```diff
@@ -1134,6 +1136,17 @@ func (c *Controller) syncHandler(workqueueItem WorkQueueItem) error {
...
+				k8sutil.ServicePortForwardPort(context.TODO(), newNamespace, "traefik", 32080, 80, color.FgGreen)
+			}
+		}
 	} else { // Case: DELETE
```
Comment (MortalHappiness) [inline submarine-cloud-v2/controller.go:1150]: You do not remove the port forward in delete event. So if the submarine custom resource is deleted and re-create again, some bug will happen.

**Review by MortalHappiness — CHANGES_REQUESTED** [review]

File: submarine-cloud-v2/controller.go (hunk 3 above)
Comment (kevin85421) [reply submarine-cloud-v2/controller.go:1152]: I have fixed it.

File: submarine-cloud-v2/controller.go (hunk 2 above)
Comment (kevin85421) [reply submarine-cloud-v2/controller.go]: I have fixed it.

**Comment by kevin85421** [comment]
@MortalHappiness  Thank you for your recommendations. I have fixed the problems.

**Review by MortalHappiness — APPROVED** [review]
LGTM

**Review by Kenchu123 — APPROVED** [review]
LGTM

**Pull request closed without merging**